### DISCUSSION TITLE
Questions about `sv1_api` crate

### DISCUSSION THREAD

**Discussion by plebhash** [stratum-mining/stratum#897] [General]
This discussion should guide the documentation of `sv1_api` crate, as described on https://github.com/stratum-mining/stratum/issues/845

Unlike most crates under `protocols`, this crate is already quite well documented.

But we have community members starting to adopt it for their projects (e.g.: @0xB10c), so this Discussion could still be a good place to clarify things and potentially come up with ideas to improve the docs.

**Comment by 0xB10C** [comment]
Any reason why to not use [BlockHash](https://docs.rs/bitcoin/latest/bitcoin/struct.BlockHash.html) from rust-bitcoin for [`sv1_api::utils::PrevHash`](https://docs.rs/sv1_api/latest/sv1_api/utils/struct.PrevHash.html). I guess to keep it generic (i.e. other cryptocurrencies?) and to not be dependent on rust-bitcoin?


Edit:

> PrevHash in Stratum V1 has brain-damaged serialization as it swaps bytes of every u32 word into big endian. Therefore, we need a special type for it

I guess it makes sense to offer the raw value (even if it's brain-damaged) to a consumer.

**Reply by Fi3** [reply]
This come from braiins open codebase. It is like that just cause was working code. And the only functionality that we needed was serialize and deserialize it (that differ from the one in BlockHash). I agree that for the user could be handy to get a bitcoin::BlockHash instead. Maybe we can implement Into BlockHash instead of substitute it, so that we don't break the API.
